src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -42,41 +42,108 @@
+ *     SymbolLookup libGL = SymbolLookup.libraryLookup("libGL.so"); // libGL.so loaded here
+ *     MemorySegment glGetString = libGL.lookup("glGetString");
+ *     ...
+ * } //  libGL.so unloaded here
+ * }
  * <p>
- * Finally, clients can load a library and obtain a {@linkplain #libraryLookup(Path, MemorySession) library lookup} which can be used
- * to find symbols in that library. A library lookup is associated with a {@linkplain  MemorySession memory session},
- * and the library it refers to is unloaded when the session is {@linkplain MemorySession#close() closed}.
+ * If a library was previously loaded through JNI, i.e., by {@link System#load(String)}
+ * or {@link System#loadLibrary(String)}, then the library was also associated with a particular class loader. The factory
+ * method {@link #loaderLookup()} creates a symbol lookup for all the libraries associated with the caller's class loader:
+ *
+ * {@snippet lang=java :
+ * System.loadLibrary("GL"); // libGL.so loaded here
+ * ...
+ * SymbolLookup libGL = SymbolLookup.loaderLookup();
+ * MemorySegment glGetString = libGL.lookup("glGetString");
+ * }
+ *
+ * This symbol lookup, which is known as a <em>loader lookup</em>, is dynamic with respect to the libraries associated
+ * with the class loader. If other libraries are subsequently loaded through JNI and associated with the class loader,
+ * then the loader lookup will expose their symbols automatically.
+ * <p>
+ * Note that a loader lookup only exposes symbols in libraries that were previously loaded through JNI, i.e.,
+ * by {@link System#load(String)} or {@link System#loadLibrary(String)}. A loader lookup does not expose symbols in libraries
+ * that were loaded in the course of creating a library lookup:
+ *
+ * {@snippet lang=java :
+ * libraryLookup("libGL.so", session).lookup("glGetString").isPresent(); // true
+ * loaderLookup().lookup("glGetString").isPresent(); // false
+ * }
+ *
+ * Note also that a library lookup for library {@code L} exposes symbols in {@code L} even if {@code L} was previously loaded
+ * through JNI (the association with a class loader is immaterial to the library lookup):
+ *
+ * {@snippet lang=java :
+ * System.loadLibrary("GL"); // libGL.so loaded here
+ * libraryLookup("libGL.so", session).lookup("glGetString").isPresent(); // true
+ * }
+ *
+ * <p>
+ * Finally, each {@link Linker} provides a symbol lookup for libraries that are commonly used on the OS and processor
+ * combination supported by that {@link Linker}. This symbol lookup, which is known as a <em>default lookup</em>,
+ * helps clients to quickly find addresses of well-known symbols. For example, a {@link Linker} for Linux/x64 might choose to
+ * expose symbols in {@code libc} through the default lookup:
+ *
+ * {@snippet lang=java :
+ * Linker nativeLinker = Linker.nativeLinker();
+ * SymbolLookup stdlib = nativeLinker.defaultLookup();
+ * MemorySegment malloc = stdlib.lookup("malloc");
```

Comment:
This should be:
```suggestion
 * Optional<MemorySegment> malloc = stdlib.lookup("malloc");
```
or
```suggestion
 * MemorySegment malloc = stdlib.lookup("malloc").orElseThrow();
```